pkg/scheduler/objects/application.go

```diff
@@ -126,8 +126,8 @@ type Application struct {
 	appEvents             *schedEvt.ApplicationEvents
 	sendStateChangeEvents bool // whether to send state-change events or not (simplifies testing)
 
-	snapshot        bytes.Buffer
-	appSnapshotLock locking.RWMutex
+	snapshot     bytes.Buffer
+	snapshotLock locking.RWMutex
```

Comment:
We can use only the basic lock. We never lock it for read so it would be much better if type also communicates it?
Also, minor styling detail: can we move the lock above the `snapshot` because the convention is to put lock above the group it guards (in this case, `snapshot`)?

Reply:
Good suggestions - fixed in commit 6fd7458.